### PR TITLE
MentorshipPage - fixed sca svg issue in mentorships page

### PR DESCRIPTION
**Pull request by arjunpotter17**
**Description**

This PR fixes the SVG issue in dark more for the she code africa logo in the mentorship page. 

**Notes for Reviewers**
To check the fix, navigate to community - handbook - mentorships-programs and toggle between light and dark mode. The issue also fixes logo issues for other articles in the same page.

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Shivam-AfA** [comment]
@arjunpotter17 Build is failing, please check the error. It seems like a path issue.

**Comment by arjunpotter17** [comment]
@leecalcote @Shivam-AfA I fixed the build, but the lighthouseseci check is failing. I tried to go through it but I'm not sure if I can make much sense of it. Can any of you help?

**Comment by arjunpotter17** [comment]
@leecalcote oh, okay! This was my first contribution. Thanks for the supprt! I apologize for any mishaps. Is there anything else that I'm supposed to do?

**Comment by arjunpotter17** [comment]
@leecalcote @Udit-takkar if any of you can help me a bit, I can take care of issue #3789 too. I'm having trouble passing the theme context into the .mdx file. Is there a workaround for this?

File: src/sections/Community/Handbook/programs-data.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
+import MLHLightlogo from "../../../collections/programs/mlh-2020/mlh-dark.svg";
 import LFX from "../../../collections/programs/lfx-2022/Linux_Foundation_logo.png";
+import LFXLightlogo from "../../../collections/programs/lfx-2022/lfx-white.svg";
 import Hacktoberfest from "../../../assets/images/programs/hacktoberfest.png";
 import GSOC from "../../../collections/programs/gsoc-2020/gsoc.svg";
 import GSOD from "../../../collections/programs/gsod/gsod.png";
 import Layer from "../../../assets/images/layer5/layer5-only/svg/layer5-white-bg.svg";
+import LayerLightlogo from "../../../assets/images/layer5/layer5-only/svg/layer5-dark-bg.svg";
 import Bridge from "../../../collections/programs/communitybridge/communitybridge-blue.png";
+import BridgeDark from "../../../collections/programs/communitybridge/communitybridge.png";
```
Comment (Udit-takkar) [inline src/sections/Community/Handbook/programs-data.js:13]: prefer using .webp format files instead of png.

**Review by Udit-takkar — APPROVED** [review]
LGTM. Great work just convert png image to webp if that doesn't affect the quality of photo

File: src/sections/Community/Handbook/programs-data.js (hunk 1 above)
Comment (arjunpotter17) [reply src/sections/Community/Handbook/programs-data.js:13]: Oh, I just used whatever files were available. I didn't add any. Only png files were available, so I used them.

Comment (Udit-takkar) [reply src/sections/Community/Handbook/programs-data.js:13]: yeah no issue just replace them with webp

**Comment by Udit-takkar** [comment]
@arjunpotter17 about this https://github.com/layer5io/layer5/issues/3789. i don't know the best method to do this but one way i found is to create a wrapper around https://github.com/layer5io/layer5/blob/master/src/collections/programs/sca-contributhon/index.mdx?plain=1#L15 and give a className to image tag then  use props.theme in the wrapper to load the correct image path using this property https://developer.mozilla.org/en-US/docs/Web/CSS/content

**Comment by leecalcote** [comment]
@arjunpotter17 were you able to get this one figured out?

**Comment by arjunpotter17** [comment]
Hey @leecalcote 

> @arjunpotter17 were you able to get this one figured out?

Hey @leecalcote , I'm sorry I replied this late. I had a personal mishap and I was busy with that. I was under the impression that #4046 was fixed. As for #3789 I remember trying to implement what @Udit-takkar suggested but it didn't work. If you could point me in the right direction, I'll work on it right away. Also, for #4046 do you want me to convert PNG to WEBP? I only used whatever images were already present in the project already and they were of PNG type itself.

**Comment by arjunpotter17** [comment]
@Udit-takkar can you pitch in? I just have to convert png to webp and we're good to go right?

**Comment by Nikhil-Ladha** [comment]
What's the blocker here? Are we waiting on something? Or, do you need some kind of assistance in taking it through @arjunpotter17 ?

**Comment by randychilau** [comment]
Hi @arjunpotter17 ,

I was hoping to provide a suggestion for this PR that may be helpful.

Recently the site updated how the site and SVGs changed between light and dark themes. Regarding SVGs specifically, the goal was to have one SVG that would be color responsive to the theme changing between light and dark mode, instead of a conditional based on a variable (e.g.` isDark ? darkIcon.svg : lightIcon.svg`). 

For the [Mentorship Programs](https://layer5.io/community/handbook/mentorship-programs) page, there are a few extra steps to get the responsive SVG working:

- import the svg as a React component into the `programs_data` file
- configure the component to replace the image
- in the `mentorship.js` file, you will need to create a conditional to detect if it is a React component or an external file (since some images in the `programs_data` file are `png`).
 note: for the `png` files you can [use a theme conditional](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#changing-images-depending-on-the-theme) for displaying the correct png.
- update the styled-components accordingly to style the `svg`

I outlined the process of creating a responsive SVG in the dropdown below.

If the merge conflicts with this current PR (from the recent code changes) are an issue with updating this PR, then perhaps the Maintainers on this thread can help or you can start over by closing this PR and creating a new one, regardless I am happy to be a resource and/or assist.

Feel free to let me know if you have any questions and good luck.

Cheers,
Randy


<details>
  <summary>Click here for info on creating a responsive SVG</summary>

In order to achieve the goal of a responsive SVG, it required a few steps:

1. creating a responsive SVG from the original SVG
2. import the SVG as a React component instead of an external file ([why?](https://weekendprojects.dev/posts/fixing-css-svg-fill-not-working/#2-make-sure-to-use-inline-svgs-instead-of-external))
3. adding css to the respective styled-component that targeted classNames in the SVG responsible for color fills, and having it change color when the theme changed color.

---

As an example, let's use the responsive Layer5 SVG logo in the header.

The first step,` creating a responsive SVG from the original SVG`:

image

The original SVG file is located `src/assets/images/app/layer5.svg`, and if you open that file you will notice:
image

There is a `style` section responsible for the color fills of the image, and we wanted the Layer 5 logo to look like this in darkmode:
image

So we will want to target `.cls-1` since that is the color we want to be responsive to theme changing.

For the sake of clarity, a separate SVG is created, and the term "colorMode" is included in the file name:
`src/assets/images/app/layer5-colorMode.svg`

and the targeted class name is edited to keep consistency with the goal and filename:

image

---

Now the second step:  `import the SVG as a React component`:

Layer5 site uses a plugin that allows importing the SVG as React component, and when the component is rendered, the svg is inlined (the actual SVG code is in the page, and not referenced like an external file), By the SVG being inlined, it can be targeted with CSS by the styled-component.

For the navigation, the file `src/sections/General/Navigation/index.js` is where we are importing the SVG as a React Component, and it is done with the following `import` statement:

`import { ReactComponent as Logo } from "../../../assets/images/app/layer5-colorMode.svg";
`
and in that file, where the original svg was used in an `img` tag, we replace it with the SVG React component.

image

---

Next is the last step: `adding css to the respective styled-component`:

For the navigation component, we see that `<NavigationWrap>` is controlling the styling of the component.

image

image

And when we go to that file `src/sections/General/Navigation/navigation.style.js`, we will want the class that is styling the SVG component

image

and add styles for a `svg` element

image

If you notice, the css targets the `.colorMode1` class name in the svg with `.layer5-colorMode_svg__colorMode1`. 

Why are they different? I believe this is because of a [feature of styled-components](https://styled-components.com/docs/basics):
>No class name bugs: styled-components generates unique class names for your styles. You never have to worry about duplication, overlap or misspellings.

These unique class names can be seen in the resulting HTML through the browser. 

image

And if we had multiple SVGs on the page that we wanted to change to different colors, by specifically targeting the generated class name, we can be sure to change each one with the correct color. (there may be a better method to handle this).

So using [styled-components theming](https://styled-components.com/docs/advanced#theming), we can use the props to grab a property where the color changes with the theme (explanation of color pairs [here](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#adding-dark-mode-for-components)). 

`fill: ${props => props.theme.whiteToGreen3C494F};`

note: always include the transition `transition: fill 0.8s cubic-bezier(0.2, 0.8, 0.2, 1);` whenever you change colors so it is in sync with the rest of the site.

So when the site is in dark mode, the targeted class name will fill the SVG parts controlled by that class name with the color `white`, and when it is in light mode, it will fill a dark green color hexcode `#3C494F`.

---

And that gets you to a responsive SVG to dark and light mode.

[logo_change.webm](https://github.com/layer5io/layer5/assets/90356410/0f0a63e0-bbb7-43a0-b8b4-c58c9f9ebf8f)

additional reference: [CONTRIBUTING.md](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#changing-svg-according-to-theme)

</details>

**Comment by Savio629** [comment]
@arjunpotter17 updates?

**Comment by Shivam-AfA** [comment]
Let's wait for another two days for @arjunpotter17's updates. Else, @randychilau will you take over these issues?

**Comment by randychilau** [comment]
Hi @Shivam-AfA,

Sure, happy to help.

Cheers,
Randy

**Comment by arjunpotter17** [comment]
Hey, @randychilau. Sorry for being so inconsistent. You can take over. I'm swamped right now. Again, sorry for the troubles guys.

**Comment by randychilau** [comment]
Hi All,

Good day.

Closing this PR and resubmitting a new PR as discussed in the above conversation.

Cheers,
Randy

**Pull request closed without merging**